src/Engine/RulesComponent.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\RulesComponent.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\rules\Context\ContextDefinitionInterface;
+
+/**
+ * Handles executable Rules components.
+ */
+class RulesComponent {
```

Comment:
Yes. We need to rename this to RulesComponentConfig. As discussed, let's do this in a follow-up.